### ISSUE TITLE
Plugin options conflict with CLI options

### ISSUE THREAD

**Issue by gempain** [semantic-release/semantic-release#1696]
Hi there - I'll skip how amazing you lib is, by now you should have realized that it should be in every dev's toolbox,

When writing plugins, I realized that the plugin options override CLI options. Take this plugin, say `test.js`:

```js
module.exports = {
  verifyConditions: (options) => {
    console.log(options);
  }
}
```

and this config `.eslintrc.js`:

```js
module.exports = {
  branches: 'latest',
  plugins: [
    ['./test', {}],
  ],
};
```

and run

```
$ semantic-release -d --ci false
...
{
  ...
  ci: false,
  noCi: true
}
```

Now update your config to:

```js
module.exports = {
  branches: 'latest',
  plugins: [
    ['./test', {ci: 'tada !'}],
  ],
};
```

And you get:

```
$ semantic-release -d --ci false
...
{
  ...
  ci: 'tada !',
  noCi: true
}
```

This could lead to issues. I think plugin options should be passed in a different option.

What do you think ?

**Comment by gr2m** [comment]
Thanks for the kind words :) 

hmm I agree, it feels wrong that CLI options get overwritten by plugin configuration ... there might be a good reason but I don't know what it could be. Could you investigate? Feel free to start a PR and we continue the discussion there?

**Comment by EvanCarroll** [comment]
I investigated a different issue @gr2m. In my case, I was wondering why `noCi` was getting set when environment `CI=1` was set. For me `get-config.js` has

```js
if (options.ci === false) {
  options.noCi = true;
}
```

This sets the `noCi` option if `ci` isn't set and it ignores `env` entirely. 

All of this logic is overly complex. Why are there two options `ci` and `noCi`? Shouldn't this simply be 

* command-line options set CI options if set
* If not set, see if environmental variable CI is set
* If not set, default CI to off in options.

**Comment by gr2m** [comment]
I agree with the sentiment. I haven't written the code, so I can't say why it was written the way. 

That being said, reducing complexity and maintenance overhead is @travi's and my top priority right now, so we'd much appreciate if anyone would like to look into it and find out if we could simplify the options without breaking common use cases

**Comment by EvanCarroll** [comment]
I simplified the options in my own branch with this,

```js
if (options.ci === true || options.ci === false) {
  options.noCi = !options.ci;
}
else if (options.noCi === true || options.noCi === false) {
  options.ci = !options.noCi;
}
else if (env["CI"] !== undefined) {
  let ci = env["CI"] ? true : false;
  options.ci = ci;
  options.noCi = !ci;
}
```

If we adopted this code the two options would always be in sync via `options.ci` because we don't document the differences anywhere (that I see), we should assume that such was always the intent. If bugs arise we can fix them. I can supply patch if desired.

**Comment by EvanCarroll** [comment]
@gr2m  patch up.
